src/plots/cartesian/axes.js

```diff
@@ -2622,6 +2630,11 @@ axes.drawOne = function(gd, ax, opts) {
             rangeSliderPush = Registry.getComponentMethod('rangeslider', 'autoMarginOpts')(gd, ax);
         }
 
+        if(typeof ax.automargin === 'string') {
+            filterPush(push, ax.automargin);
+            filterPush(mirrorPush, ax.automargin);
```

Comment:
How about `rangeSliderPush`? Should we filter that one too?

Reply:
ha @nickmelnikov82 asked me the same question, my answer:

> that one isn’t controlled by `ax.automargin` so I think no

There are others that aren't even in this section of code, like legends and colorbars. `ax.automargin` is only about "are the axes themselves allowed to increase the margins."

Reply:
Oh that's interesting... in principle I like this, one place to control all of this rather than requiring you to set it in each axis separately. But it opens a can of worms: by default axes have `automargin: false` while colorbars and legends (and rangesliders) by default DO increase the margins, and putting this at the top level would imply that it applies to everything. So it would need a default value like `'legacy'` or something that gets inherited differently by axes and other objects, and then we would need to extend support for disabling automargin to all these other objects as well.

All that to say: it's a good idea, but it's going to be a bunch of work, so let's not do it now. What we're doing in this PR would be fully compatible with adding that sometime in the future.